### PR TITLE
Setup build dev process

### LINKED ISSUE #8: setup build / dev process
* [ ] Webpack - build
* [ ] Webpack - dev
* [ ] Live reload - dev

### PR DESCRIPTION
**Pull request by jennschiffer**
resolves #8 

* sets up webpack, react, babel
* uses webpack-dev-server as a server/watcher of files
* updates readme with information on how to run/build this skeletal app :skull:

### CODE REVIEW

File: src/app.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+* @file The main Nest Service application
+*/
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+/* Load index.html */
+import index from 'file?name=[name].[ext]!./index.html';
```
Comment (cowboy) [inline src/app.jsx]: Can you create a `/\.html$/` loader in the webpack config so this import can be simplified thus?

```js
import index from './index.html';
```

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+var path = require('path');
+
+module.exports = {
+  context : path.join(__dirname, './'),
+  entry: './src/app.jsx',
```
Comment (cowboy) [inline webpack.config.js]: If the trailing `.jsx` is removed, will this still work?

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  resolve: {
+    extensions: ['', '.js', '.jsx', '.html']
+  },
+  module: {
+    loaders: [
+      {
+        test: /\.jsx?$/,
+        loader: 'babel-loader',
+        exclude: /node_modules/,
```
Comment (cowboy) [inline webpack.config.js]: I'm not sure this `exclude` is necessary. Can it be removed?

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        query: {
+          presets: ['es2015', 'react']
```
Comment (cowboy) [inline webpack.config.js]: Is it possible to load this stuff from the `.babelrc` file? What happens if it's omitted?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,26 @@
-# nest2
-NEW AND IMPROVED
+# Service Catalog (Nest, part deux)
```
Comment (cowboy) [inline README.md]: Can you add a section to the README that's just a bullet-list of the npm scripts, with a super brief description of each script? We'll be adding more!

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+node_modules
```
Comment (cowboy) [inline .gitignore]: Prefix root-level directory names with `/` (this and `public`)

File: webpack.config.js (hunk 2 above)
Comment (jennschiffer) [inline webpack.config.js]: yep!

File: webpack.config.js (hunk 4 above)
Comment (jennschiffer) [inline webpack.config.js]: it looks like the default `cacheIdentifier` loader property is set to pull .babelrc - it seems to work when i omit the query property (see: https://github.com/babel/babel-loader#options)

File: webpack.config.js (hunk 3 above)
Comment (jennschiffer) [inline webpack.config.js]: yeah, now that i think about it, node_modules isn't in the src folder so yep!

File: src/app.jsx (hunk 1 above)
Comment (jennschiffer) [inline src/app.jsx]: i have to specify the name to copy it over to or else it creates a hash for the name and gets all borked up. so reading up on this, it seems that the general consensus is this is not a webpack thing to do really (everyone is like "use grunt or gulp").

Comment (cowboy) [inline src/app.jsx]: How about we use the `babel-cli` and `fs-extra` npm modules and create this script:

### tools/build-html.js:

```js
import fs from 'fs-extra';

fs.copy('src/index.html', 'public/index.html');
```

And add the following npm scripts:

```js
"build:html": "babel-node tools/build-html",
"prebuild": "npm run build:html",
```

And remove all the fancy `index.html` stuff from the JS code?

(PS. untested)

Comment (jennschiffer) [inline src/app.jsx]: just updated and this works great!

**Comment by jennschiffer** [comment]
if this works for you, feel free merge @cowboy!

**Comment by cowboy** [comment]
@jennschiffer if I delete `public` then `npm start` doesn't work. Does webpack need to be made aware of `index.html` or the `src` directory?

File: package.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "service-catalog",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "build": "webpack",
+    "start": "webpack-dev-server --progress --colors --content-base public",
```
Comment (cowboy) [inline package.json]: Change `public` to `src` so the webpack server can find `index.html`.

Comment (jennschiffer) [inline package.json]: \o/

**Pull request merged**